patches/api/0480-Add-event-so-you-can-modify-what-players-will-hear-t.patch

```diff
@@ -0,0 +1,111 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lurkrul <44719592+Lurkrul@users.noreply.github.com>
+Date: Wed, 19 Jun 2024 17:57:27 +0800
+Subject: [PATCH] Add event so you can modify what players will hear the event
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/DragonPrepareDeathSoundTargetsEvent.java b/src/main/java/io/papermc/paper/event/entity/DragonPrepareDeathSoundTargetsEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..c9f821dd291c486c56597cc32135f481592daeff
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/DragonPrepareDeathSoundTargetsEvent.java
+@@ -0,0 +1,99 @@
++package io.papermc.paper.event.entity;
++
++import org.bukkit.entity.Player;
++import org.bukkit.entity.EnderDragon;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++import java.util.Collection;
++import java.util.HashSet;
++import java.util.Set;
++
++/**
++ * Is called when an {@link EnderDragon} dies to determine what {@link Player}s hear it.
++ */
++public class DragonPrepareDeathSoundTargetsEvent extends EntityEvent {
++
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++    private Set<Player> playerList;
++
++    public DragonPrepareDeathSoundTargetsEvent(final @NotNull EnderDragon enderDragon, Collection<? extends Player> players) {
++        super(enderDragon);
++        playerList = new HashSet<>(players);
++    }
++
++    /**
```

Comment:
Either expose the collection directly and remove the other modifying accessors or return a copy of that Set and drop the setPlayerList(Set<....).  Having both will just unnecessarily bloat the event with methods.

Reply:
removed all of the functions and changed access of the set to publci